kpi-tracker/src/main.rs

```diff
@@ -0,0 +1,248 @@
+
+                new_accounts.insert(address, account_details);
+            }
+            _ => {}
+        }
+    }
+
+    Ok(new_accounts)
+}
+
+/// Insert as a single DB transaction to facilitate easy recovery, as the service can restart from
+/// current height stored in DB.
+fn update_db(
+    db: &mut DB,
+    (block_hash, block_details): (BlockHash, BlockDetails),
+    accounts: BTreeMap<AccountAddress, AccountDetails>,
+) {
+    db.blocks.insert(block_hash, block_details);
+    db.accounts.extend(accounts.into_iter());
+}
+
+async fn handle_block(
+    node: &mut v2::Client,
+    block: FinalizedBlockInfo,
+    db: &mut DB,
+) -> anyhow::Result<()> {
+    let block_info = node
+        .get_block_info(block.block_hash)
+        .await
+        .context(format!(
+            "Could not get block info for block: {}",
+            block.block_hash
+        ))?
+        .response;
+
+    let block_details = BlockDetails {
+        block_time: block_info.block_slot_time,
+        height: block_info.block_height,
+    };
+
+    let new_accounts = if block.height.height == 0 {
+        accounts_in_block(node, &block.block_hash).await?
+    } else {
+        new_accounts_in_block(node, &block_info.block_hash, &db.accounts).await?
+    };
+
+    update_db(db, (block_info.block_hash, block_details), new_accounts);
+
+    Ok(())
+}
+
+async fn use_node(endpoint: v2::Endpoint, height: AbsoluteBlockHeight) -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    println!("Using node {}\n", endpoint.uri());
+
+    let mut db = DB {
+        blocks: HashMap::new(),
+        accounts: HashMap::new(),
+    };
+
+    let mut node = Client::new(endpoint)
+        .await
+        .context("Could not connect to node.")?;
+
+    let mut blocks_stream = node
+        .get_finalized_blocks_from(height)
+        .await
+        .context("Error querying blocks")?;
+
+    for _ in 0..args.num_blocks {
+        if let Some(block) = blocks_stream.next().await {
+            handle_block(&mut node, block, &mut db).await?;
+        }
+    }
+
+    println!("\n");
+    println!("Blocks stored: {}\n", &db.blocks.len());
+
+    let mut accounts: Vec<(AccountAddress, DateTime<Utc>, AccountDetails)> = db
+        .accounts
+        .into_iter()
+        .map(|(address, details)| {
+            let block_time = db
+                .blocks
+                .get(&details.block_hash)
+                .expect("Found account with wrong reference to block?")
+                .block_time;
+
+            (address, block_time, details)
+        })
+        .collect();
+
+    accounts.sort_by_cached_key(|v| v.1);
+
+    let account_strings: Vec<String> = accounts
+        .into_iter()
+        .map(|(address, block_time, details)| format!("{}, {}, {:?}", address, block_time, details))
+        .collect();
+
+    println!("Accounts stored:\n{}", account_strings.join("\n"));
+
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() {
```

Comment:
```suggestion
async fn main() -> anyhow::Result<() {
```
This way you can use `?` inside `main` which makes error handling simpler.